core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -784,41 +810,35 @@ public InodeLockList lockDescendant(LockedInodePath inodePath, LockMode lockMode
    * @param inodePath the root {@link LockedInodePath} to retrieve all descendants from
    * @param lockMode the lock type to use
    * @return an {@link InodeLockList} representing the list of all descendants
-   * @throws FileDoesNotExistException if inode does not exist
    */
-  public InodeLockList lockDescendants(LockedInodePath inodePath, LockMode lockMode)
-      throws FileDoesNotExistException {
-    Inode<?> inode = inodePath.getInode();
-    InodeLockList inodeGroup = new InodeLockList();
-    if (!inode.isDirectory()) {
-      return inodeGroup;
-    }
-    return lockDescendantsInternal((InodeDirectory) inode, lockMode, inodeGroup);
+  public LockedInodePathList lockDescendants(LockedInodePath inodePath, LockMode lockMode) {
+    List<LockedInodePath> inodePathList = new ArrayList<>();
+    lockDescendantsInternal(inodePath, lockMode, inodePathList);
```

Comment:
Is there any chance that this throws an exception? Then we could have some orphan `LockedInodePath`s which are never closed.

Reply:
Should not throw an exception, since we catch the exceptions in `lockDescendantsInternal` and handle them.